File: app/src/main/java/eu/warble/pjapp/data/StudentDataSource.java
package eu.warble.pjapp.data;


import eu.warble.pjapp.data.model.Student;

public interface StudentDataSource {

    interface LoadStudentDataCallback {
        void onDataLoaded(Student studentData);
        void onDataNotAvailable(String error);
    }

    void getStudentData(LoadStudentDataCallback callback);
}

File: app/src/main/java/eu/warble/pjappkotlin/data/StudentDataSource.kt
package eu.warble.pjappkotlin.data

import android.content.Context
import eu.warble.pjappkotlin.data.model.Student

interface StudentDataSource {

    interface LoadStudentDataCallback {
        fun onDataLoaded(studentData: Student)
        fun onDataNotAvailable(error: String)
    }

    fun getStudentData(appContext: Context, callback: LoadStudentDataCallback)
}